x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/TransportGetAsyncSearchAction.java

```diff
@@ -57,8 +59,12 @@ protected void doExecute(Task task, GetAsyncSearchAction.Request request, Action
                         ActionListener.wrap(
                             p -> getSearchResponseFromTask(searchId, request, nowInMillis, expirationTime, listener),
                             exc -> {
-                                if (exc.getCause() instanceof DocumentMissingException == false) {
-                                    logger.error("failed to retrieve " + searchId.getEncoded(), exc);
+                                //don't even log when: the async search document or its index is not found. That can happen if an invalid
+                                //search id is provided and no async search initial response has been stored yet.
+                                if (exc.getCause() instanceof DocumentMissingException == false
+                                    && exc.getCause() instanceof IndexNotFoundException == false) {
```

Comment:
I was wondering if we are sure about exc.getCause here. What's the top level exception?

Reply:
A `RemoteTransportException`.  Maybe we can replace the instanceof with `ExceptionsHelper.status(exc) != RestStatus.NOT_FOUND` ?

Reply:
the problem I have with `ExceptionsHelper.status(exc)` is that it does not look at the cause at all and it gives 500 if it does not know any better. How about `ExceptionsHelper.status(ExceptionsHelper.unwrapCause(exc))` ?

Reply:
+1